measurements/api/private.py

```diff
@@ -44,34 +44,49 @@ def api_private_stats_by_month(orm_stat):
     end_date = datetime(now.year, now.month, 1)
     start_date = end_date - relativedelta(months=24)
 
-    r = current_app.db_session.query(
-            func.date_trunc('month', Report.test_start_time).label('test_start_month'),
-            orm_stat
-        ).filter(Report.test_start_time >= start_date).filter(Report.test_start_time < end_date
-        ).group_by('test_start_month')
+    s = select([
+        func.date_trunc('month', sql.text('bucket_date')).label('bucket_month'),
+        sql.text(orm_stat)
+    ]).where(and_(
+        sql.text("bucket_date > :start_date"),
+        sql.text("bucket_date < :end_date")
+    )).group_by(
+        'bucket_month'
+    ).select_from(sql.table('ooexpl_bucket_msm_count'))
+
+    r = current_app.db_session.execute(s, {'start_date': start_date, 'end_date': end_date})
     result = [{
         'date': (bkt + relativedelta(months=+1, days=-1)).strftime("%Y-%m-%d"),
         'value': value,
     } for bkt, value in sorted(r)]
-    return jsonify(result)
+    return result
 
 @api_private_blueprint.route('/asn_by_month')
 def api_private_asn_by_month():
-    # The query takes ~6s on local SSD @ AMS on 2018-04-04.
-    # It was taking ~45s when it was fetching all the table from DB and doing grouping locally.
-    return api_private_stats_by_month( func.count(distinct(Report.probe_asn)) )
```

Comment:
I see that you are removing these comments but cannot guess why you're doing it now

Reply:
Because the query time is no longer so slow (I have done optimisations on it) so the comment is no longer relevant.